sw/host/cryptotest/testvectors/parsers/random_ecdsa_generator.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import logging
+import math
+import sys
+
+import jsonschema
+from Crypto.Hash import SHA256, SHA384, SHA512, SHA3_256, SHA3_384, SHA3_512
+from Crypto.PublicKey import ECC
+from Crypto.Signature import DSS
+from cryptotest_util import rng
+
+MESSAGE_LENGTH = 512
+
+
+def generate_test_vectors(args):
+    test_vectors = []
+    random = rng()
+    # Parse tests within the group
+    for count in range(args.count):
+        # Generate random ECC key pair
+        key_pair = ECC.generate(curve=args.curve, randfunc=random)
+        public = key_pair.pointQ
+        msg = random(MESSAGE_LENGTH)
+        signature = []
+        # Sign digest of message using private key
+        if args.hash_alg == "sha-256":
+            digest = SHA256.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha-384":
+            digest = SHA384.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha-512":
+            digest = SHA512.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha3-256":
+            digest = SHA3_256.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha3-384":
+            digest = SHA3_384.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha3-512":
+            digest = SHA3_512.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        else:
+            raise ValueError("Unsupported hash algorithm " + args.hash_alg)
+
+        signature = list(signature)
+
+        verify_result = True
+        # Decide whether to flip a bit in the signature, creating a
+        # negative test.
+        if random(1)[0] % 2 == 0:
+            if len(signature) == 0:
+                # To change an empty signature, make it a single random byte
+                signature = list(random(1))
+            else:
+                # Flip a random bit in the signature
+                idx = int.from_bytes(random(4), "big") % (len(signature) * 8)
+                signature[math.floor(idx / 8)] ^= 1 << (idx % 8)
+                verify_result = False
+
+        r = signature[:int(len(signature) / 2)]
```

Comment:
nit: you can do the integer division with `//` (here and on the line below)
```suggestion
        r = signature[:len(signature) // 2]
```

Reply:
A new operator a day keeps the `ValueError`s away :)